### DISCUSSION TITLE
Error while creating new user with prisma adapter

### DISCUSSION THREAD

**Discussion by marsidorowicz** [nextauthjs/next-auth#4516] [Help] [answered]
### Adapter type

@next-auth/prisma-adapter

### Environment

  System:
    OS: Windows 10 10.0.19044
    CPU: (6) x64 Intel(R) Core(TM) i5-9600K CPU @ 3.70GHz    
    Memory: 7.20 GB / 31.92 GB
  Binaries:
    Node: 16.13.0 - C:\Program Files\nodejs\node.EXE
    npm: 8.1.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (101.0.1210.32)
    Internet Explorer: 11.0.19041.1566
  npmPackages:
    next: ^11.1.3 => 11.1.3 
    next-auth: ^4.3.4 => 4.3.4 
    react: ^17.0.2 => 17.0.2 

  npmPackages:
    @next-auth/prisma-adapter: ^1.0.3 => 1.0.3

### Reproduction URL

https://github.com/nextauthjs/next-auth-example

### Describe the issue

Seems like one data object in AccountUncheckedCreateInput is missing:
data.created_at

Invalid `p.account.create()` invocation in

  16 },
  17 updateUser: ({ id, ...data }) => p.user.update({ where: { id }, data }),
  18 deleteUser: (id) => p.user.delete({ where: { id } }),
→ 19 linkAccount: (data) => p.account.create({
       data: {
           xxx
       }
     })

Unknown arg `created_at` in data.created_at for type AccountUncheckedCreateInput. Available args:

type AccountUncheckedCreateInput {
  id?: String
  userId: String
  type: String
  provider: String
  providerAccountId: String
  refresh_token?: String | Null
  access_token?: String | Null
  expires_at?: Int | Null
  token_type?: String | Null
  scope?: String | Null
  id_token?: String | Null
  session_state?: String | Null
}

 Error:
Invalid `p.account.create()` invocation in

  16 },
  17 updateUser: ({ id, ...data }) => p.user.update({ where: { id }, data }),
  18 deleteUser: (id) => p.user.delete({ where: { id } }),
→ 19 linkAccount: (data) => p.account.create({
       data: {
         provider: 'coinbase',
         type: 'oauth',

       }
     })

Unknown arg `created_at` in data.created_at for type AccountUncheckedCreateInput. Available args:

type AccountUncheckedCreateInput {
  id?: String
  userId: String
  type: String
  provider: String
  providerAccountId: String
  refresh_token?: String | Null
  access_token?: String | Null
  expires_at?: Int | Null
  token_type?: String | Null
  scope?: String | Null
  id_token?: String | Null
  session_state?: String | Null
}

  name: 'LinkAccountError',
  code: undefined
}

### How to reproduce

add coinbase provider with credentials
login using coinbase
use database prisma adapter

### Expected behavior

adapter should add Account along with User

**Comment by balazsorban44** [comment, marked as answer]
created_at is not part of the default schema, but you can add it if you need it. 👍

**Reply by marsidorowicz** [reply]
so to solve it a position in Object Account prisma model (or any) needs to be added:
created_at        Int?

then npx prisma db push and npx prisma generate, after this it works, so the main model needs to be replaced to be:

```
model Account {
  id                String  @id @default(cuid())
  userId            String
  type              String
  created_at        Int?
  provider          String
  providerAccountId String
  refresh_token     String?
  access_token      String?
  expires_at        Int?
  token_type        String?
  scope             String?
  id_token          String?
  session_state     String?
  user              User    @relation(fields: [userId], references: [id], onDelete: Cascade)

  @@unique([provider, providerAccountId])
}
```

**Reply by IsaiahByDayah** [reply]
Just ran into this same issue creating a site that uses the built-in [PatreonProvider](https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/providers/patreon.ts).

I ended up having to add a `version String?` field to my [Account model](https://next-auth.js.org/adapters/prisma#create-the-prisma-schema), run `yarn prisma migrate dev`, and then things started working. In the event that a provider supplies fields that aren't in the default Account, is it usually "safe" to follow the solution here? (I'm still fairly new to next-auth / prisma / databases so sorry if this is an obvious question)

Perhaps it'd be good to add a page to the docs outlining any 1-off Account models you might need to account for depending on the provider used?

**Reply by marsidorowicz** [reply]
All of this depends on Adapter and there are so many so devs probably don't catch up with all fields in Model

**Comment by Y80** [comment]
![image](https://user-images.githubusercontent.com/40132433/194046259-4f7e36db-1575-40ec-970f-82d67b0543d9.png)

**Reply by Y80** [reply]
Manually delete property `created_at`。

Also, u can add other properties there.

**Comment by Salahuddin-Gazi** [comment]
linkAccount: (data) => {
  // Modify the expires_at key to expiresAt
  const modifiedData = { ...data, expiresAt: data.expires_at };
  delete modifiedData.expires_at;

  return p.account.create({ data: modifiedData });
},
